profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/compute/ProfitBricksComputeServiceAdapter.java

```diff
@@ -0,0 +1,570 @@
+            final org.apache.jclouds.profitbricks.rest.domain.Volume.Request.CreatePayload request = org.apache.jclouds.profitbricks.rest.domain.Volume.Request.creatingBuilder()
+                    .dataCenterId(dataCenterId)
+                    .image(image.getId())
+                    .imagePassword(password)
+                    .name(format("%s-disk-%d", name, i++))
+                    .size(volume.getSize().intValue())
+                    .type(VolumeType.HDD)
+                    .build();
+
+            String volumeId = (String) provisioningManager.provision(jobFactory.create(dataCenterId, new Supplier<Object>() {
+
+               @Override
+               public Object get() {
+                  return api.volumeApi().createVolume(request).id();
+               }
+            }));
+
+            volumeIds.add(volumeId);
+            logger.trace(">> provisioning complete for volume. returned id='%s'", volumeId);
+         } catch (Exception ex) {
+            if (i - 1 == 1) // if first volume (one with image) provisioning fails; stop method
+            {
+               throw Throwables.propagate(ex);
+            }
+            logger.warn(ex, ">> failed to provision volume. skipping..");
+         }
+      }
+
+      String volumeBootDeviceId = Iterables.get(volumeIds, 0); // must have atleast 1
+      waitVolumeUntilAvailable.apply(VolumeRef.create(dataCenterId, volumeBootDeviceId));
+
+      // provision server
+      final String serverId;
+      Double cores = ComputeServiceUtils.getCores(hardware);
+
+      Server.BootVolume bootVolume = Server.BootVolume.create(volumeBootDeviceId);
+
+      try {
+         final Server.Request.CreatePayload serverRequest = Server.Request.creatingBuilder()
+                 .dataCenterId(dataCenterId)
+                 .name(name)
+                 .bootVolume(bootVolume)
+                 .cores(cores.intValue())
+                 .ram(hardware.getRam())
+                 .build();
+
+         logger.trace("<< provisioning server '%s'", serverRequest);
+
+         serverId = (String) provisioningManager.provision(jobFactory.create(dataCenterId, new Supplier<Object>() {
+
+            @Override
+            public Object get() {
+               return api.serverApi().createServer(serverRequest).id();
+            }
+         }));
+         logger.trace(">> provisioning complete for server. returned id='%s'", serverId);
+
+      } catch (Exception ex) {
+         logger.error(ex, ">> failed to provision server. rollbacking..");
+         destroyVolumes(volumeIds, dataCenterId);
+         throw Throwables.propagate(ex);
+      }
+
+      waitServerUntilAvailable.apply(ServerRef.create(dataCenterId, serverId));
+      waitDcUntilAvailable.apply(dataCenterId);
+
+      //attach bootVolume to Server
+      api.serverApi().attachVolume(Server.Request.attachVolumeBuilder()
+              .dataCenterId(dataCenterId)
+              .serverId(serverId)
+              .volumeId(bootVolume.id())
```

Comment:
IIRC, if one creates a server (line 193) with the `bootVolume` param set, it'll be automatically attached after that server is provisioned. Is this not the case anymore?